### PR TITLE
fix headers TrackPropagation/RungeKutta

### PR DESCRIPTION
**Pull request by mrodozov**
Fix headers related to #15248

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

File: TrackPropagation/RungeKutta/src/CartesianStateAdaptor.h [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #ifndef CartesianStateAdaptor_H
 #define CartesianStateAdaptor_H
 
+#include "RKSmallVector.h"
```
Comment (perrotta) [inline TrackPropagation/RungeKutta/src/CartesianStateAdaptor.h]: Better to include headers with the full path
(I see that this is already not always satisfied by the current code: @mrodozov do you want to propagate it, at least into the files touched by this pull request?)

File: TrackPropagation/RungeKutta/src/RKCartesianDerivative.h [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #ifndef RKCartesianDerivative_H
 #define RKCartesianDerivative_H
 
-class CartesianState;
+#include "CartesianState.h"
```
Comment (perrotta) [inline TrackPropagation/RungeKutta/src/RKCartesianDerivative.h]: Better to include headers with the full path

File: TrackPropagation/RungeKutta/src/RKCartesianDistance.h [hunk 3]
```diff
@@ -2,6 +2,7 @@
 #define RKCartesianDistance_H
 
 #include "FWCore/Utilities/interface/Visibility.h"
+#include "DataFormats/GeometryVector/interface/Basic3DVector.h"
 #include "RKDistance.h"
```
Comment (perrotta) [inline TrackPropagation/RungeKutta/src/RKCartesianDistance.h]: e.g.

File: TrackPropagation/RungeKutta/src/RKCurvilinearDistance.h [hunk 4]
```diff
@@ -1,6 +1,7 @@
 #ifndef RKCurvilinearDistance_H
 #define RKCurvilinearDistance_H
 
+#include "DataFormats/GeometryVector/interface/Basic3DVector.h"
 #include "FWCore/Utilities/interface/Visibility.h"
 #include "RKDistance.h"
```
Comment (perrotta) [inline TrackPropagation/RungeKutta/src/RKCurvilinearDistance.h]: e.g.

File: TrackPropagation/RungeKutta/src/RKOne4OrderStep.h [hunk 5]
```diff
@@ -2,6 +2,7 @@
 #define RKOne4OrderStep_H
 
 #include "FWCore/Utilities/interface/Visibility.h"
+#include "TrackPropagation/RungeKutta/src/RKDistance.h"
 #include "RK4OneStepTempl.h"
```
Comment (perrotta) [inline TrackPropagation/RungeKutta/src/RKOne4OrderStep.h]: e.g.

File: TrackPropagation/RungeKutta/src/CartesianStateAdaptor.h (hunk 1 above)
Comment (makortel) [reply TrackPropagation/RungeKutta/src/CartesianStateAdaptor.h]: @perrotta I agree in general (i.e. for headers in `interface`), but why would it be better for headers in `src` (or `plugins`) that ought to be implementation details of the library (plugin) and not to be `#include`'d elsewhere?

File: TrackPropagation/RungeKutta/src/RKCartesianDerivative.h (hunk 2 above)
Comment (VinInn) [reply TrackPropagation/RungeKutta/src/RKCartesianDerivative.h]: not for local header

**Comment by perrotta** [comment]
Thank you @makortel and @VinInn 

Indeed, at the end my concern was probably more aesthetical than practical (since sometime the full path was written, e.g. in https://github.com/mrodozov/cmssw/blob/328080eeba3882fd5513d2f340ca5244e2dca142/TrackPropagation/RungeKutta/src/RKCartesianDistance.h#L5, and sometime it is not)

I will not insist on it, unless @mrodozov wants to profit and make uniform the includes in the files touched by this PR

File: TrackPropagation/RungeKutta/src/RKCartesianDerivative.h (hunk 2 above)
Comment (slava77) [reply TrackPropagation/RungeKutta/src/RKCartesianDerivative.h]: I agree: local headers (not in interface/) should be included like this.
And whoever is including A/B/src/B.h from a package C/D is doing it wrong.

**Comment by mrodozov** [comment]
oops I messed it, I'll close it and propose another one with the changes for the headers

**Comment by perrotta** [comment]
@mrodozov : I think the conclusion of the discussion  (that I started) was to NOT include the full path.
Therefore, the only one to be fixed is the only one where you actually already did it:
/blob/328080eeba3882fd5513d2f340ca5244e2dca142/TrackPropagation/RungeKutta/src/RKOne4OrderStep.h#L5

**Pull request closed without merging**